### DISCUSSION TITLE
Multi-Cluster Management at the Edge

### DISCUSSION THREAD

**Discussion by rstoermer** [k3s-io/k3s#11434] [Q&A]
What are you guys using to manage your clusters at the Edge? We have been using Portainer for some time but have had some hiccups. Given that we might manage >100 K3s clusters at the edge, what are options to:
- Update the underlying operating system
- Create a reverse tunnel if no SSH access is possible due to a Firewall
- Update K3s
- Monitor applications from a unified Dashboard in the Cloud and maybe automatically update K3s from the Cloud
- ...

Thanks!

**Comment by vasdee** [comment]
I have similar questions and some of what I posted in https://github.com/k3s-io/k3s/discussions/11431 might be relevant information here. 

In terms of OS, i've been looking a little bit into flat car or talos. Then using the immutable layer nature to progressively roll out updates - including k3s itself. 

This would require firewall rules and i imagine some hosting of some kind, haven't looked into that as yet. I'm hopeful OCI could be used for those artifacts
